src/extension.ts

```diff
@@ -100,10 +104,70 @@ const processCustomEvent = async (e: vscode.DebugSessionCustomEvent): Promise<an
   }
 };
 
+const lookForModules = (source: string): Object => {
+  const rm = /^(var|let|const)?\s*([a-zA-Z0-9$_]+)\s*=[\s|\n]*require\s*\(\s*['"]([a-zA-Z0-9$_]+)['"]\s*\);?$/;
+  return source.split('\n').filter(line => rm.test(line)).map(m => {
+    const match = rm.exec(m);
+
+    return {
+      link: match[2],
+      mod: match[3]
+    };
+  });
+};
+
+const createItems = (document: vscode.TextDocument, position: vscode.Position): Array<vscode.CompletionItem> => {
+  const rm = /([a-zA-Z0-9 =]+)\.$/;
+  const items = new Array<vscode.CompletionItem>();
+  const modules = Object.keys(iotjs);
+  const textUntilPos = document.getText(document.getWordRangeAtPosition(position));
+  const currentModule = lookForModules(textUntilPos);
+  const might = (document.lineAt(position.line).text).split(/\s/g).pop().replace(/\./g, '');
+  const matchKey = Object.keys(currentModule).find(key => currentModule[key].link === might);
+
+  if (document.lineAt(position.line).text.match(rm)) {
+    modules.forEach(mod => {
+      if (document.lineAt(position.line).text.includes(currentModule[matchKey].link) &&
+        currentModule[matchKey].mod === mod) {
+        for (let i in iotjs[mod]) {
+          items.push(new vscode.CompletionItem(iotjs[mod][i].label, 2));
+          items[i].detail = iotjs[mod][i].detail;
+          items[i].insertText = iotjs[mod][i].insertText;
+          items[i].documentation = iotjs[mod][i].documentation;
+        }
+      }
+    });
+  }
+  return items;
+};
+
+const createModules = (document: vscode.TextDocument, position: vscode.Position): Array<vscode.CompletionItem> => {
+  const rm = /^(var|let|const)?\s*([a-zA-Z0-9$_]+)\s*=[\s|\n]*require\s*\(\s*['"]/;
+  const moduleItems = new Array<vscode.CompletionItem>();
+  const modules = Object.keys(iotjs);
+
+  if (document.lineAt(position.line).text.match(rm)) {
+    for (let i in modules) {
+      moduleItems.push(new vscode.CompletionItem(modules[i], 2));
+    }
+  }
+  return moduleItems;
```

Comment:
I would use the following pattern here: (because we need default modules that always availabe in iotjs no matter they are required or not)
```javascript
// at the top of the file
const defaultModules = [{
  link: 'process',
  mod: 'process',
}, {
  link: 'emitter',
  mod: 'events',
}, {
  link: 'timers',
  mod: 'timers',
}];

// in the createModules
if (document.lineAt(position.line).text.match(rm)) {
  return [...defaultModules, ...modules.map(key => new vscode.CompletionItem(modules[key], 2))];
}

return defaultModules;
```

In this case we do not need the following line:
```javascript
const moduleItems = new Array<vscode.CompletionItem>();
```
The `createItems` should be similar because we have to deal with the default modules.